templates/admin/actions.html

```diff
@@ -0,0 +1,28 @@
+{% load i18n %}
+
+{% if action_form %}
+  <div class="actions">
+    {% for field in action_form %}
+      {% if field %}
+        <div class="action-field">{{ field.label }} {{ field }}</div>
+      {% endif %}
+    {% endfor %}
+
+    <div class="button-action">
+      <button type="submit" class="button small" title="{% trans "Run the selected action" %}" name="index" value="{{ action_index|default:0 }}">
```

Comment:
how does this button get the index? is it something that is built in into Django admin functionality?

Reply:
I think it is from the action form and passed here within context but it will be always None or 0 since we do not have query params or any default action specified

Reply:
As far as I know it is used only as default value

Reply:
Hm... I just don't like having code about which I don't understand how it works... I see similar code snippets on SO and GitHub, but there is no explanation.

Reply:
> I think it is from the action form and passed here within context but it will be always None or 0 since we do not have query params or any default action specified

Yes, you're right:

> The `value` attribute holds the data that will be sent when the form is submitted. In this case, it's using the `action_index` variable from the Django context. If `action_index` is not provided in the context, it defaults to `0`. This value helps identify which action was selected when the form is processed on the server side.
> 
> The `action_index` is a variable that should be passed to the template context by the Django view that renders this template. It's not defined within the template itself; it's part of the context data provided by the view. If action_index is not included in the context, the default filter sets it to `0` as a fallback. This ensures that the form can still be submitted without errors even if `action_index` is missing.